titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/common/grpc/AbstractTitusGrpcServer.java

```diff
@@ -71,7 +77,9 @@ public int getPort() {
     @PostConstruct
     public void start() {
         if (!started.getAndSet(true)) {
-            ServerBuilder serverBuilder = configure(ServerBuilder.forPort(configuration.getPort()));
+            ServerBuilder<?> initial = ServerBuilder.forPort(port);
+            initial.executor(ExecutorsExt.instrumentedCachedThreadPool(runtime.getRegistry(), "grpcCallbackExecutor"));
```

Comment:
How are these executors shutdown?

Reply:
Never, only when the JVM shuts down. The executor will live with the gRPC server (one per gRPC server).

Reply:
Hi, I would recommend that you shut down the gRPC server, even if you don't bother with the executor.  You can add it as a JVM shutdown hook.    This allows in flight RPCs to finish gracefully while turning away new client.

Reply:
We run multiple server instances (on ephemeral ports) when running integration tests. Servers are shutdown properly, so it should be easy to add a shutdown logic to the executor pool as well.

Reply:
sounds good, I will look into tearing down the executor when the grpc server is shutting down in a follow-up PR

Reply:
@tbak I read the code again, and I was wrong and misinterpreted the code. You were right, by default it seems each gRPC *server* gets a single cached thread pool. gRPC *clients* get a thread pool per `ManagedChannel`. I mixed up the two usages of the `callbackExecutor` code.

In any case I still think it is better to replace the default server callback executor, as we will get better instrumentation with our own. I will make a change to avoid leaking executors when we spawn multiple grpc servers in integration tests in a follow-up PR.